ld-pruning-wf.wdl

```diff
@@ -0,0 +1,440 @@
+task ld_pruning {
+	input {
+		File gds
+		File? sample_include_file
+		File? variant_include_file
+		String genome_build = "hg38"  # can also be hg18 or hg19
+		Float ld_r_threshold = 0.32
+		Float ld_win_size = 10  # yes, a float, not an int
+		Float maf_threshold = 0.01
+		Float missing_threshold = 0.01
+		#Boolean autosome_only = true  # buggy and not in CWL
+		Boolean exclude_pca_corr = true  # will act as String in Python
+		String? out_prefix
+		
+		# runtime attributes
+		Int addldisk = 1
+		Int cpu = 2
+		Int memory = 4
+		Int preempt = 3
+	}
+	command {
+		set -eux -o pipefail
+
+		echo "Generating config file"
+		# noquotes = argument has no quotes in the config file to match CWL config EXACTLY
+		python << CODE
+		import os
+		f = open("ld_pruning.config", "a")
+		f.write('gds_file "~{gds}"\n')
+		if "~{exclude_pca_corr}" != "true":
+			f.write("exclude_pca_corr ~{exclude_pca_corr}\n")  # noquotes
+
+		# to match CWL exactly, genome_build is written even if default value
+		if "~{genome_build}" == "hg38":
+			f.write('genome_build "~{genome_build}"\n')
+		else:
+			if ("~{genome_build}" == "hg18" or "~{genome_build}" == "hg19"):
+				f.write('genome_build "~{genome_build}"\n')
+			else:
+				f.close()
+				print("Invalid ref genome. Please only select either hg38, hg19, or hg18.")
+				exit(1)
+		
+		if ~{ld_r_threshold} != 0.32:
+			f.write("ld_r_threshold ~{ld_r_threshold}\n")  # noquotes
+		if ~{ld_win_size} != 10:  # by default this compares 10.0 to 10 which evals to true
+			f.write("ld_win_size ~{ld_win_size}\n")  # noquotes
+		if ~{maf_threshold} != 0.01:
+			f.write("maf_threshold ~{maf_threshold}\n")  # noquotes
+		if ~{missing_threshold} != 0.01:
+			f.write("missing_threshold ~{missing_threshold}\n")  # noquotes
+
+		if "~{def_sampleInc}" == "true":
+			f.write('sample_include_file "~{sample_include_file}"\n')
+
+		if "~{def_variantInc}" == "true":
+			f.write('variant_include_file "~{variant_include_file}"\n')
+
+		if "chr" in "~{gds}":
+			parts = os.path.splitext(os.path.basename("~{gds}"))[0].split("chr")
+			outfile_temp = "pruned_variants_chr" + parts[1] + ".RData"
+		else:
+			outfile_temp = "pruned_variants.RData"
+		if "~{out_prefix}" != "":
+			outfile_temp = "~{out_prefix}" + "_" + outfile_temp
+
+		f.write('out_file "' + outfile_temp + '"\n')
+
+		f.close()
+		CODE
+
+		echo "Calling R script ld_pruning.R"
+		Rscript /usr/local/analysis_pipeline/R/ld_pruning.R ld_pruning.config
+	}
+	
+	# Estimate disk size required
+	Int gds_size = ceil(size(gds, "GB"))
+	Int sample_size = if defined(sample_include_file) then ceil(size(sample_include_file, "GB")) else 0
+	Int varInclude_size = if defined(variant_include_file) then ceil(size(variant_include_file, "GB")) else 0
+	Int finalDiskSize = gds_size + sample_size + varInclude_size + addldisk
+
+	# Workaround for optional files
+	Boolean def_sampleInc = defined(sample_include_file)
+	Boolean def_variantInc = defined(variant_include_file)
```

Comment:
Can you put this above the command section?

Also, probably good to stick to underscore instead of a combination of underscore and camelCase.

Reply:
`defined()` is a WDL command that doesn't work properly in the command section. Where it's currently located makes it basically a private input variable.

I will fix the casing though.